test/position-managers/PositionManager.t.sol

```diff
@@ -71,8 +72,8 @@ contract PositionManagerTest is Test, PosmTestSetup, LiquidityFuzzers {
         PositionConfig memory config = PositionConfig({poolKey: key, tickLower: 0, tickUpper: 60});
         bytes memory calls = getMintEncoded(config, 1e18, address(this), "");
 
-        // SafeTransferLib does not bubble the ContractLocked error and instead reverts with its own error
-        vm.expectRevert("TRANSFER_FROM_FAILED");
+        // Permit2.transferFrom does not bubble the ContractLocked error and instead reverts with its own error
+        vm.expectRevert(abi.encodeWithSelector(IAllowanceTransfer.AllowanceExpired.selector, 0));
```

Comment:
vm.expectRevert("TRANSFER_FROM_FAILED");
mmm this is no longer reverting for reentrancy problems if its reventing with allowance expired. To test reentrancy you need to give permit2 allowance to the reentrant token - so that permit2 doesnt prevent the transfer. It should still be failing with `TRANSFER_FROM_FAILED` from solmate

Reply:
If you run this test with -vvvvv youll see in the stack trace that theres never reentrancy because permit2 rejects the transfer before it reenters

Reply:
https://github.com/Uniswap/v4-periphery/pull/208/commits/d18599ef5bc41f3af76b29f48003fe2b0b687968

proof of fix:
![Screenshot 2024-07-29 at 18 50 54](https://github.com/user-attachments/assets/503fdc8f-a55c-4422-9773-a1f94a207cd3)